src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.Worker.cs

```diff
@@ -98,10 +99,12 @@ private void ProcessNodes()
                 {
                     _cancellationToken.ThrowIfCancellationRequested();
                     var nodeOrToken = _pendingNodes.Pop();
+                    var intervalsThatIntersect = _textSpanIntervalTree.GetIntervalsThatIntersectWith(
+                            nodeOrToken.FullSpan.Start, nodeOrToken.FullSpan.Length);
```

Comment:
```suggestion
                        nodeOrToken.FullSpan.Start, nodeOrToken.FullSpan.Length);
```

Reply:
this is costly to do in a loop.  if you actually need to literally pass this data down (unclear if that's needed), then create a temp buffer and call *Add*IntervalsThatIntersectWith.  Then clear the buffer on each loop.